riscv/decode.h

```diff
@@ -330,6 +330,23 @@ do { \
                         ((STATE.prv == PRV_U) && (s##field == 0)))) \
       throw trap_virtual_instruction(insn.bits()); \
   } while (0);
+#define require_stateen_fcsr \
```

Comment:
This is an unfortunate copy/paste from `fcsr_csr_t::verify_permissions()`, which is a public member. Can you simply call `STATE.fflags->verify_permissions()` here?

On a related note, shouldn't the code in `fcsr_csr_t::verify_permissions()` belong in `float_csr_t::verify_permissions()`? It seems that the sstaten bit should apply to both `frm` and `fflags` and not only the composite `fcsr`, right?

Reply:
I'm not sure about whether stateen.FCSR bit should apply to frm and fflags which is not stated in the spec. They doesn't currently. And fcsr is not an element of state. So I copied the check logic here or Maybe I need add fcsr to state or add verification_csr function to search CSR_FCSR from csrmap and do the verify_permissions

Reply:
fflags and frm are part of fcsr, so anything that controls access to fcsr must also control access to fflags and frm.

Reply:
OK. I agree with this. I'll send a fix to this.

Reply:
PR #1059 fixes this